### PR TITLE
LUCENE-9589 Swedish Minimal Stemmer

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/LUCENE-9589 moved to new repo, see also old PR at https://github.com/apache/lucene-solr/pull/2062

### CODE REVIEW

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/*
+ * This algorithm is updated based on code located at:
+ * http://members.unine.ch/jacques.savoy/clef/
+ *
+ * Full copyright for that code follows:
+ */
+
+/*
+ * Copyright (c) 2005, Jacques Savoy
```
Comment (janhoy) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java:27]: Could these copyright notices be moved to LICENSE.txt or NOTICE.txt ?

**Comment by janhoy** [comment]
@rmuir  you already reviewed this PR over at lucene-solr repo. I am still trying to get feedback from a native swede but, otherwise I'm ready to merge this aiming for 8.9..

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java (hunk 1 above)
Comment (HoustonPutman) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java:27]: I believe they should be moved to the NOTICE.txt

**Comment by karlwettin** [comment]
I gave the stemmer a spin on [SAOL](https://en.wikipedia.org/wiki/Svenska_Akademiens_ordlista) 13 (2006). I have to stay within the bounds of fair use and can't publish the complete results.

Generally speaking I think it does a remarkable job with such a small decision tree. Given what it's meant to do, I would merge it.

A few notes that are more applicable on a not so minimal implementation:

The suffix-s pluralis rule have ~5300 exceptions where words ends with s is nominative case singularis.

It's however missing the rules defined in LUCENE-1515, especially 'an' and 'ans'-suffixes. Back then I came to the conclusion that 8% of the Swedish language can be inflected that way, but there is a list of ~200 words that needs to be setup as exceptions to those rules.

Two standard an/ans-suffixes:

| Stemmed        | Original           |
| ------------- |:-------------:|
ättiksgurk | ättiksgurka
ättiksgurka | ättiksgurkan
ättiksgurka | ättiksgurkans
ättiksgurk | ättiksgurkas
ättiksgurk | ättiksgurkor
ättiksgurk | ättiksgurkorna
ättiksgurk | ättiksgurkornas
ättiksgurk | ättiksgurkors

| Stemmed        | Original           |
| ------------- |:-------------:|
ättestup | ättestupa
ättestupa | ättestupan
ättestupa | ättestupans
ättestup | ättestupas
ättestup | ättestupor
ättestup | ättestuporna
ättestup | ättestupornas
ättestup | ättestupors

There are probably more complete and better examples of this in LUCENE-1515.

And if I have to go looking for problems, I see these:

| Stemmed        | Original           |
| ------------- |:-------------:|
höstmörk | höstmörker
höstmörk | höstmörkers
höstmörkr | höstmörkret
höstmörkr | höstmörkrets

| Stemmed        | Original           |
| ------------- |:-------------:|
höstkollektio | höstkollektion
höstkollektion | höstkollektionen
höstkollektion | höstkollektionens
höstkollektion | höstkollektioner
höstkollektion | höstkollektionerna
höstkollektion | höstkollektionernas
höstkollektion | höstkollektioners
höstkollektio | höstkollektions

This one is a number of different words with very different meaning that turn out completely mixed up, not all nous though:

| Stemmed        | Original           |
| ------------- |:-------------:|
hölj | hölj
hölj | hölja
hölja | höljan
höljand | höljande
hölja | höljans
hölj | höljas
höljd | höljd
höljd | höljda
höljd | höljde
höljd | höljdes
hölj | hölje
hölj | höljen
höljen | höljena
höljen | höljenas
hölj | höljens
hölj | höljer
hölj	höljes
hölj	höljet
hölj	höljets
hölj	höljor
hölj	höljorna
hölj	höljornas
hölj	höljors
hölj	höljs
höljt	höljt
höljt	höljts

I'm afraid it isn't possible to extract stemmer rules and exception lists from SAOL due to copyright issues (unless we find a digital copy that's at least 20 years old), but perhaps an alternative and more global route would be to mine [Wikidata:Lexicographical data](https://www.wikidata.org/wiki/Wikidata:Lexicographical_data)?

https://www.wikidata.org/wiki/Lexeme:L38829

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java (hunk 1 above)
Comment (janhoy) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/sv/SwedishMinimalStemmer.java:27]: Hmm, I see a ton of other stemmers with exactly the same headers. So I'll leave them as is in this PR and rather do a separate copyright cleanup.

**Comment by janhoy** [comment]
Since release 8.9 is in feature freeze I now target this at 9.0.0. I moved CHANGES entry and @since tags. Will commit later this week.

**Pull request merged**